html_css_dom/the_internet/README.md

```diff
@@ -0,0 +1,183 @@
+- `301` which means "this was moved elsewhere". If you type `google.com` and your browser takes you to `www.google.com` its because of a `301` response.
+- `404` which means "this page was not found"
+- `500` which means "the server encountered an error". Sometimes you get a nice message back about why, but often you do not.
+
+**HTTP Request Methods**
+
+There are multiple types of requests that clients can make. 
+
+These are known sometimes as HTTP "verbs". You can think of each one as an action.
+
+* GET - give me some data
+* POST - take this data
+* PUT - update this data
+* DELETE - delete this data
+
+
+## Making your own requests
+
+You may not realize that you have already been making requests!
+
+Every time you load a web page, some requests are made. We can look at all of the requests using the browser's devtools network tab.
+
+* Open any website, like the [main curriculum page](https://github.com/joinpursuit/Pursuit-Core-Web)
+* Open the devtools (Cmd-Shift-i)
+* Click the `Network` tab
+* Refresh the page
+
+![network tab](./assets/network-tab.png)
+
+You'll see a whole bunch of stuff happening as the page loads!
+
+### Anatomy of a request
+
+Each line here is a request that the browser makes. At the bottom is a summary - this page took 38 requests and 11.46 seconds to load everything!
+
+The first request is to get the actual HTML for the page. From there, the browser parses the HTML and then makes a request for each resource that the HTML contains - one for each image, script, css file, font, and anything else the HTML has. There might also be other requests - these are triggered by the javascript that runs on the page.
+
+If you look at the columns, you can see useful information about each request. The `Status` and `Type` are probably the most helpful here.
+
+Status is the `HTTP status code` like we talked about above. `Type` tells you what the response contains. 
+
+> What do you think all the different `Type` values mean?
+
+If you click on a row, you'll see more detailed information about that request and response. The main thing to look at here is the `Headers` tab. Each request contains descriptive information telling the server what to expect. You can see the request `Method` like we talked about above, and lots of other information that probably doesn't make a ton of sense right now. We'll get into what some of the headers mean in later lessons.
+
+### You do: Different website requests
+
+Pick 3 different websites that you know. Type them into the URL bar and see how many network requests each one takes to load.
+
+Scroll through the requests and see if anything comes out at you that surprises you. Write it down! We'll talk about it together.
+
+## Okay let's actually make requests
+
+Hopefully you have [Insomnia](https://insomnia.rest/download/) installed already. If not, take a minute to download and install Insomnia Core.
```

Comment:
Separately from this, should we make `Download Insomnia` an assignment or task in Canvas with a due date?

Reply:
Insomnia was listed as part of the initial setup stuff, so most everyone should have it installed

Reply:
Right! I forgot that